lib/serviceClient.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * provides a service client that can be subclassed by specific
+ * implementations, as well as other helpers useful when writing services.
+ *
+ *This is intended for the CLI use case, so errors aren't as
+ * helpful as they could be.  If we want to separate this into a library,
+ * we'll need to clean it up some.
+ *
+ * To use, subclass this class, pass the parameters up to the constructor,
+ * and then use this.request to make requests. The class will handle
+ * authenticating and manual url location.
+ */
+const request = require('request');
+const urlJoin = require('url-join');
+
+class ServiceClient {
```

Comment:
I really like the idea of standarizing things like that, however I'm against that without team discussion. Also `serviceClient` name suggests a name for `service client`  - something that all of our services can use to communicate with any type of service, but this class has a specific logic to your service.